### PR TITLE
Share modal

### PR DESCRIPTION
**Pull request by CodexAdrian**

### CODE REVIEW

File: lib/components/base/ShareModal.vue [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+const sendTweet = computed(
+  () =>
+    `https://twitter.com/intent/tweet?text=` +
+    encodeURIComponent(`${props.shareText}\n${content.value}`)
+)
+
+const sendToot = computed(
+  () =>
+    `https://mastodon.social/share?text=` +
```
Comment (triphora) [inline lib/components/base/ShareModal.vue:79]: mastodon.social isn't the only Mastodon instance. Needs to have a prompt to enter the instance to use.

File: lib/components/base/ShareModal.vue [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+const qrCode = ref(null)
+const qrImage = ref(null)
+const content = ref(null)
+
+const show = async (passedContent) => {
+  content.value = passedContent
+  if (props.link) {
+    const url =
+      'https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=' +
```
Comment (triphora) [inline lib/components/base/ShareModal.vue:45]: Surely we don't need to use an external API for this. Quick google search returns https://github.com/cozmo/jsQR

**Review by triphora — DISMISSED** [review]

File: lib/components/base/ShareModal.vue [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+</template>
+
+<style scoped lang="scss">
+.share-body {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  flex-wrap: wrap;
+  gap: var(--gap-sm);
+  padding: var(--gap-xl);
```
Comment (Geometrically) [inline lib/components/base/ShareModal.vue:194]: this is too much padding, see other modals for the correct amount

File: docs/.vitepress/config.js [hunk 4]
```diff
@@ -44,6 +44,7 @@ export default {
           { text: 'Markdown', link: '/components/markdown' },
           { text: 'Copy Code', link: '/components/copy-code' },
           { text: 'Notifications', link: '/components/notifications' },
+          { text: 'ShareModal', link: '/components/share-modal' },
```
Comment (Geometrically) [inline docs/.vitepress/config.js:47]: this should be "Share Modal"

**Review by Geometrically — CHANGES_REQUESTED** [review]
needs to be more padding here 
image


What does the clipboard button do?

image
This seems weird to style as an input, since you can't edit it.

The

**Pull request merged**